### ISSUE TITLE
workflow.time_completed is not always populated on ctrl-C

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#1589]
**Describe the bug**
When ctrl-C is pressed to stop a workflow, the monitoring workflow.time_completed and workflow_duration fields are sometimes not properly populated.

I think this correlates with parsl hanging at ctrl-C time instead of shutting down properly in situations where the processes which receive the final WORKFLOW_INFO message from DFK shut down have already been killed by that ctrl-C.

**To Reproduce**
Do the above a few times.

**Expected behavior**
The time_completed and workflow_duration fields should be populated.

**Environment**
 - OS: [e.g. ubuntu, centos, MacOS, windows]
 - Python version
 - Parsl version

**Distributed Environment**
- Where are you running the Parsl script from ? [e.g. Laptop/Workstation, Login node, Compute node]
- Where do you need the workers to run ? [e.g. Same as Parsl script, Compute nodes, Cloud nodes]

**Comment by ZhuozhaoLi** [comment]
If you press ctrl+c multiple times, you can probably see parsl hangs on this line? https://github.com/Parsl/parsl/blob/6273ec0a55a27720e41d04ce2b29e5a9eb348f6c/parsl/dataflow/dflow.py#L933

I think one potential solution is to add an `atexit_cleanup` like the one in dfk to monitoring.

**Comment by benclifford** [comment]
yeah hanging on that send is pretty common - the send won't return unless the receiver is still running, I think.
